avocado/core/spawners/common.py

```diff
@@ -20,9 +27,47 @@ class BaseSpawner:
 
     METHODS = []
 
+    def bytes_from_file(self, filename):
+        """Read bytes from a files in binary mode.
+
+        This is a helpful method to read *local* files bytes efficiently.
+
+        If the spawner that you are implementing needs access to local file,
+        feel free to use this method.
+        """
+        # This could be optimized in the future.
+        with open(filename, 'rb', 0) as fp:
+            with mmap(fp.fileno(), 0, access=ACCESS_READ) as stream:
+                yield stream.read()
+
+    def stream_output(self, job_id, task_id):
+        """Returns output files streams in binary mode from a task.
+
+        This method will find for output files generated by a task and will
+        return a generator with tuples, each one containing a filename and
+        bytes.
+
+        You need to provide in your spawner a `stream_output()` method if this
+        one is not suitable for your spawner. i.e: if the spawner is trying to
+        access a remote output file.
+        """
+        src = '~/avocado/job-results/{}/test-results/{}/data'.format(job_id,
+                                                                     task_id)
+        try:
+            for path in Path(src).expanduser().iterdir():
+                if path.is_file():
+                    # Only files with something
+                    if path.stat().st_size == 0:
+                        continue
```

Comment:
what do you think about:

```
if path.is_file() and path.stat().st_size != 0:
```

It should reduce the code complexity of codeclimate.

Anyway, just a suggestion, won't change the logic.

Reply:
But will break the logic, because the stream loop right bellow will not be executed, right?

I will try to fix this.

Reply:
What do you mean by 'will not be executed'?

If I understood your code correctly, you execute the loop on line 63 if `path` is a file and if it has content. Otherwise, you jump to the next file.

The code I suggested is doing the same, if `path` is a file and it has content, it execute the loop inside.

Reply:
The stream loop would be executed for dirs, right? If path.is_file() return false, will not continue and will execute the stream loop. The logic would be:

```python
if not path.is_file() or path.stat().st_size == 0:
   continue
```

Reply:
Ohhh I just got your `!=` :) Yes, we are doing the same!

Reply:
@willianrampazzo thanks, just force pushed here.